lib/core/sync_job_runner.rb

```diff
@@ -54,108 +74,135 @@ def execute
     private
 
     def do_sync!
-      Utility::Logger.info("Claiming a sync job for connector #{@connector_settings.id}.")
-
-      # connector service doesn't support multiple jobs running simultaneously
-      raise Core::JobAlreadyRunningError.new(@connector_settings.id) if @connector_settings.running?
-
-      Core::ElasticConnectorActions.update_connector_last_sync_status(@connector_settings.id, Connectors::SyncStatus::IN_PROGRESS)
-
-      # claim the job
-      @job.make_running!
-
-      job_description = @job.es_source
-      job_id = @job.id
-      job_description['_id'] = job_id
-
-      unless job_id.present?
-        Utility::Logger.error("Failed to claim the job for #{@connector_settings.id}. Please check the logs for the cause of this error.")
-        return
-      end
+      return unless claim_job!
 
       begin
-        Utility::Logger.debug("Successfully claimed job for connector #{@connector_settings.id}.")
+        @sink = Core::Ingestion::EsSink.new(@index_name, @connector_settings.request_pipeline)
```

Comment:
Yes, this was missed when I resolve the conflicts. I have updated it.

Initially, I moved the `@sink` instantiation here, because there are many cases in that the job will not run, e.g. job claim fails, configuration validation fails.